lib/components/molecules/AtomicCarousel/AtomicCarousel.js

```diff
@@ -0,0 +1,341 @@
+   *
+   * @param {Object} event
+   *   keyup Object
+   */
+  const handleMouseLeave = () => {
+    if (mouseStatus) {
+      setAnimation(true);
+    }
+  };
+
+  /**
+   * Slider Section.
+   */
+  const getSliderSection = () => {
+    return (
+      <div className="carousel-container">
+        {data.length !== 0 && (
+          <>
+            {navigation && (
+              <div className="action previous">
+                <Button
+                  data-action="prev"
+                  className={classnames({
+                    'prev-icon': prevIcon,
+                  })}
+                  onClick={handlePrev}
+                  disabled={currentSlide === 1}
+                  onKeyUp={handlePreviousKey}
+                  title={prevLabel}
+                  type="button"
+                >
+                  <Icon path={prevIcon} alt="previous" title="previous" />
+                </Button>
+              </div>
+            )}
+            <div
+              className={classnames('carousel-slides', {
+                fadeIn: fade && isAnimationRunning,
+              })}
+              onMouseEnter={handleMouseEnter}
+              onMouseLeave={handleMouseLeave}
+            >
+              {getPerFrameItems(data, perSlideFrame, currentSlide)}
+            </div>
+            {navigation && (
+              <div className="action next">
+                <Button
+                  className={classnames({ 'next-icon': nextIcon })}
+                  data-action="next"
+                  title={nextLabel}
+                  onClick={handleNext}
+                  disabled={slideCount === currentSlide}
+                  onKeyUp={handleNextKey}
+                  type="button"
+                >
+                  <Icon path={nextIcon} alt="next" title="next" />
+                </Button>
+              </div>
+            )}
+          </>
+        )}
+      </div>
+    );
+  };
+
+  const getHeader = () => {
+    return (
+      <header id="carousel">
+        <Heading {...headers} className="carousel-heading">
+          {heading}
+        </Heading>
+      </header>
+    );
+  };
+
+  const ariaLiveUpdate = () => {
+    return (
+      <div className="hidden" aria-live="polite" aria-atomic="true" tabIndex="-1">
+        {`Item ${currentSlide} of ${slideCount}`}
+      </div>
+    );
+  };
+
+  const getAnimationPanel = () => {
+    return (
+      <div className="animation-panel" aria-label="animation">
+        <Button type="button" onClick={handleAnimation}>
+          <Icon
+            path={
+              (playIcon && !isAnimationRunning ? playIcon : false) ||
+              (pauseIcon && isAnimationRunning ? pauseIcon : false)
+            }
+            alt="next"
+            title="next"
+          />
+        </Button>
+      </div>
+    );
+  };
+
+  return (
+    <>
+      <section className={classnames('carousel', className)} aria-labelledby="carousel">
+        {displayHeader && getHeader()}
```

Comment:
This is want i want to avoid by adopting declarative style of programming. In this current approach the user of carousel will have no control to change any of this sequence of structure. But of you would have followed declarative use will have complete control at the structure of components

https://codeburst.io/declarative-vs-imperative-programming-a8a7c93d9ad2